cmd/govim/main.go

```diff
@@ -217,14 +220,19 @@ type govimplugin struct {
 	parentCallArgs []string
 }
 
-func newplugin(goplspath string, goplsEnv []string, defaults, user *config.Config) *govimplugin {
+func newplugin(goplspath string, goplsEnv []string, defaults, user *config.Config) (*govimplugin, error) {
 	if goplsEnv == nil {
 		goplsEnv = os.Environ()
 	}
+	// Match the logic within plugin/govim.vim
 	tmpDir := getEnvVal(goplsEnv, "TMPDIR")
 	if tmpDir == "" {
 		tmpDir = os.TempDir()
 	}
+	tmpDir = filepath.Join(tmpDir, "govim")
+	if err := os.MkdirAll(tmpDir, 0777); err != nil {
```

Comment:
Wouldn't it make sense to create one per user? Otherwise we might end up with a `/tmp/govim` directory owned by me and permissions set to 755 where no other user can write. E.g. if I have `umask` set to `022`/`0022` (common default value).

Reply:
Goodness me, despite our conversation about `/run/$UID/...` I completely forgot about this. Yes, you're absolutely right - we _have_ to create one per user. So that would end up looking like `$TMP/govim_$UID/...`. 

So now I'm left wondering whether a subdirectory is a better idea than hidden log files or not? 

BTW in my experience locally, the shutdown fixes resolve all of the tidying up issues for socket files.

Reply:
I'm going to mark this conversation as resolved, and ask that we continue it over in #1063